cmake/GenHeaderFromFile.cmake

```diff
@@ -0,0 +1,108 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include(CMakeParseArguments)
+
+# Generate a C++ header file from a text or binary file.
+#
+# The header file will contain a uint8_t array and will be generated in
+#
+#  ${CMAKE_CURRENT_BINARY_DIR}\DIR\ARRAY_NAME.hpp
+#
+# The array name is build from FILE_PATH by using the file name and replacing `-`, and `.` by `_`.
+# Example usage in CMakeLists.txt:
+#
+#   add_library(fonts_target SHARED)
+#   gen_header_from_file(TARGET fonts_target FILE_PATH `fonts\Roboto-Bold.ttf` TYPE binary)
+#
+# This will generate the `fonts\roboto_bold_ttf.hpp` in the current binary dir. The file contains an
+# array named `roboto_bold_ttf`. The file added to the project and the current binary dir is added
+# to the include paths.
```

Comment:
@AndreasHeumann  could you please comment on the motivation for this approach? Not necessarily a blocker, but trying to understand whether this has general application outside of `holoscan.slang` processing.